dali/imgcodec/parsers/jpeg2000.cc

```diff
@@ -13,18 +13,61 @@
 // limitations under the License.
 
 #include "dali/imgcodec/parsers/jpeg2000.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+
+using block_type_t = std::array<uint8_t, 4>;
+const block_type_t jp2_sig_type = {'j', 'P', ' ', ' '};
+const block_type_t jp2_format_type = {'f', 't', 'y', 'p'};
+const block_type_t jp2_header_type = {'j', 'p', '2', 'h'};
+const block_type_t jp2_im_header_type = {'i', 'h', 'd', 'r'};
+
+void advance_one_block(InputStream &stream, const block_type_t &expected_type) {
+  const auto block_size = ReadValueBE<uint32_t>(stream);
+  const auto block_type = stream.ReadOne<block_type_t>();
+  DALI_ENFORCE(block_type == expected_type, "Unexpected block type");
+
+  // Skipping the rest of the block
+  stream.Skip(block_size - sizeof(block_size) - sizeof(block_type));
+}
+
+}  // namespace
+
 ImageInfo Jpeg2000Parser::GetInfo(ImageSource *encoded) const {
+  auto stream = encoded->Open();
+
+  advance_one_block(*stream, jp2_sig_type);
+  advance_one_block(*stream, jp2_format_type);
+
+  // JPEG2000 header starts
+  stream->Skip<uint32_t>();  // block size skipped
+  DALI_ENFORCE(stream->ReadOne<block_type_t>() == jp2_header_type,
+               "Expected block type 'jp2h'");
+  stream->Skip<uint32_t>();  // image block size skipped
+  DALI_ENFORCE(stream->ReadOne<block_type_t>() == jp2_im_header_type,
+               "Expected block type 'ihdr'");
+
+  const int h = ReadValueBE<uint32_t>(*stream);
+  const int w = ReadValueBE<uint32_t>(*stream);
+  const int c = ReadValueBE<uint16_t>(*stream);
```

Comment:
```suggestion
  const int64_t h = ReadValueBE<uint32_t>(*stream);
  const int64_t w = ReadValueBE<uint32_t>(*stream);
  const int64_t c = ReadValueBE<uint16_t>(*stream);
```
uint32_t can overflow. I doubt we would ever see an image with >=2^31 size, but it's theoretically possible. This is dealing with untrusted data, so we should not assume validity nor even benevolence.